tests/adapter/dbt/tests/adapter/utils/base_array_utils.py

```diff
@@ -0,0 +1,22 @@
+from dbt.tests.adapter.utils.base_utils import BaseUtils
+from dbt.tests.util import run_dbt, check_relations_equal, get_relation_columns
+
+
+class BaseArrayUtils(BaseUtils):
+    def assert_columns_equal(self, project, expected_cols, actual_cols):
+        assert (
+            expected_cols == actual_cols
+        ), f"Type difference detected: {expected_cols} vs. {actual_cols}"
+
+    def test_expected_actual(self, project):
+        run_dbt(["build"])
+
+        # check contents equal
+        check_relations_equal(project.adapter, ["expected", "actual"])
+
+        # check types equal
+        expected_cols = get_relation_columns(project.adapter, "expected")
+        actual_cols = get_relation_columns(project.adapter, "actual")
+        print(f"Expected: {expected_cols}")
```

Comment:
do we need to print these?

Reply:
Not strictly necessary. Copied from [here](https://github.com/dbt-labs/dbt-core/blame/main/tests/adapter/dbt/tests/adapter/utils/data_types/base_data_type_macro.py#L23-L34).

Having this printing in here is particularly handy if there is a test failure in the future. Definitely came in handy for me during development too.

Reply:
In general leaving print statements in can result in debug logs being full of extraneous statements that you then have to sift through so I'd recommend removing it (or just commenting it out so if someone wants to use it to debug they can just remove the comments)

Reply:
I think `print()` statements in `pytest` tests only go to go to standard output (stdout) in two cases:
1. when there is a test failure
1. if you add the [`--show-capture` command-line option](https://docs.pytest.org/en/7.1.x/how-to/capture-stdout-stderr.html) (like `pytest ... -s`)

Otherwise, `pytest` "captures" all output so that you don't see anything other than the test summary.